custom_metrics/structured-data.js

```diff
@@ -0,0 +1,269 @@
+            (element) => element.getAttribute(attribute)
+          );
+        }
+
+        function gatherStructuredData(d) {
+          let target = {
+            jsonld_scripts: [],
+            present: {
+              json_ld: false,
+              microdata: false,
+              rdfa: false,
+              microformats2: false,
+              microformats_classic: false,
+              dublin_core: false,
+              twitter: false,
+              facebook: false,
+              opengraph: false,
+            },
+            microdata_itemtypes: [],
+            rdfa_vocabs: [],
+            rdfa_prefixes: [],
+            rdfa_typeofs: [],
+            microformats2_types: [],
+            microformats_classic_types: [],
+            meta_tags: [],
+            link_tags: [],
+            dublin_core: [],
+            twitter: [],
+            facebook: [],
+            opengraph: [],
+          };
+
+          // JSON-LD
+          target.jsonld_scripts = [
+            ...d.querySelectorAll('script[type="application/ld+json"i]'),
+          ].map((script) => script.innerHTML);
+          target.present.json_ld = target.jsonld_scripts.length > 0;
+
+          // Microdata
+          target.present.microdata =
+            d.querySelectorAll("[itemscope],[itemtype],[itemprop]").length > 0;
+          target.microdata_itemtypes = [
+            ...d.querySelectorAll("[itemtype]"),
+          ].map((element) => element.getAttribute("itemtype"));
+
+          // RDFa
+          target.present.rdfa =
+            d.querySelectorAll(
+              "[vocab],[typeof],[property],[resource],[prefix]"
+            ).length > 0;
+          target.rdfa_vocabs = getAttributeValues(d, "vocab");
+          target.rdfa_prefixes = getAttributeValues(d, "prefix");
+          target.rdfa_typeofs = getAttributeValues(d, "typeof");
+
+          // microformats2
+          [
+            "h-adr",
+            "h-card",
+            "h-entry",
+            "h-event",
+            "h-feed",
+            "h-geo",
+            "h-item",
+            "h-listing",
+            "h-product",
+            "h-recipe",
+            "h-resume",
+            "h-review",
+            "h-review-aggregate",
+          ].forEach((name) => {
+            let items = d.querySelectorAll('[class~="' + name + '" i]');
+            if (items.length > 0) {
+              target.present.microformats2 = true;
+              target.microformats2_types.push({
+                name: name,
+                count: items.length,
+              });
+            }
+          });
+
+          // Classic Microformats
+          [
+            "hAtom",
+            "hCalendar",
+            "hCard",
+            "hListing",
+            "hMedia",
+            "hProduct",
+            "hRecipe",
+            "hResume",
+            "hReview",
+            "hReview-aggregate",
+            "adr",
+            "geo",
```

Comment:
These may lead to a lot of false positives. Is there any way to be more confident that `class="geo"` uses microformats?